packages/block-library/src/gallery/edit.js

```diff
@@ -464,6 +468,22 @@ export default compose( [
 			resizedImages,
 		};
 	} ),
+	withDispatch( ( dispatch, { clientId }, { select } ) => {
+		return {
+			wasBlockJustInserted() {
+				const { clearLastBlockInserted } = dispatch( 'core/editor' );
+				const { wasBlockJustInserted } = select( 'core/editor' );
+
+				const result = wasBlockJustInserted( clientId );
+
+				if ( result ) {
+					clearLastBlockInserted();
+					return true;
+				}
+				return false;
+			},
+		};
+	} ),
```

Comment:
I repeated this `withDispatch` logic in three places. I am not sure how to make this reusable hence the duplication. Instinctively, I thought of hooks but these are class components so they are incompatible and the nature of this PR is not to do this conversion for these components. Let me know if you have any thoughts on this.

Reply:
I also wondered if this logic could somehow live within the store, but I wasn't sure about this either.